src/main/java/com/github/shyiko/mysql/binlog/BinaryLogClient.java

```diff
@@ -934,12 +925,13 @@ private void confirmSupportOfChecksum(ChecksumType checksumType) throws IOExcept
         eventDeserializer.setChecksumType(checksumType);
     }
 
-    private void listenForEventPackets() throws IOException {
+    private void listenForEventPackets(final PacketChannel channel) throws IOException {
         ByteArrayInputStream inputStream = channel.getInputStream();
         boolean completeShutdown = false;
         try {
-            while (inputStream.peek() != -1) {
+            while (!abortRequest && inputStream.peek() != -1) {
```

Comment:
Will using a `volatile` type here affect performance drastically?

Reply:
I wasn't familiar with `volatile` so I had to look it up. It looks like it's useful for making sure that data changes are reflected across multiple threads. Given that the keep alive thread doesn't check this, is `volatile` even necessary here? Am I missing another set of Threads maybe?